ddsrecorder/test/blackbox/mcap/McapFileCreationTest.cpp

```diff
@@ -0,0 +1,713 @@
+    }
+
+    return send_data;
+}
+
+mcap::LinearMessageView get_msgs_mcap(
+        std::string file_name,
+        mcap::McapReader& mcap_reader_)
+{
+    auto status = mcap_reader_.open(file_name);
+
+    auto messages = mcap_reader_.readMessages();
+
+    return messages;
+}
+
+std::tuple<unsigned int, double> record_with_transitions(
+        std::string file_name,
+        McapHandlerState init_state,
+        unsigned int first_round,
+        unsigned int secound_round,
+        McapHandlerState current_state,
+        bool event,
+        unsigned int time_sleep = 0,
+        unsigned int downsampling = 1)
+{
+    {
+        // Create Publisher
+        create_publisher(
+            test::topic,
+            test::domain,
+            DataTypeKind::HELLO_WORLD);
+
+        // Create Recorder
+        std::shared_ptr<eprosima::ddsrecorder::participants::McapHandler> mcap_handler;
+
+        std::unique_ptr<core::DdsPipe> recorder;
+        if (init_state == McapHandlerState::STOPPED)
+        {
+            recorder = create_recorder(file_name, mcap_handler, downsampling, McapHandlerState::RUNNING);
+            usleep(100000);
+            mcap_handler->stop();
+        }
+        else
+        {
+            recorder = create_recorder(file_name, mcap_handler, downsampling, init_state);
+        }
+
+        // Send data
+        for (unsigned int i = 0; i < first_round; i++)
+        {
+            send_sample();
+        }
+
+        if (init_state != current_state)
+        {
+            switch (current_state)
+            {
+                case McapHandlerState::RUNNING:
+                    mcap_handler->start();
+                    break;
+                case McapHandlerState::STOPPED:
+                    mcap_handler->stop();
+                    break;
+                case McapHandlerState::PAUSED:
+                    mcap_handler->pause();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (!time_sleep)
+        {
+            time_sleep = rand() % 2;
+        }
+        sleep(time_sleep);
+
+        for (unsigned int i = 0; i < secound_round; i++)
+        {
+            send_sample();
+        }
+
+        if (event && init_state == McapHandlerState::PAUSED)
+        {
+            if (init_state == current_state)
+            {
+                mcap_handler->trigger_event();
+            }
+        }
+    }
+
+    mcap::McapReader mcap_reader;
+    auto messages = get_msgs_mcap(file_name, mcap_reader);
+
+    unsigned int n_received_msgs = 0;
+    uint64_t actual_time = std::chrono::duration_cast<std::chrono::nanoseconds>
+                (std::chrono::high_resolution_clock::now().time_since_epoch()).count();
+    double max_timestamp = 0;
+    for (auto it = messages.begin(); it != messages.end(); it++)
+    {
+        n_received_msgs++;
+        double time_seconds = ((actual_time) - (it->message.publishTime)) * pow(10.0, -9.0);
```

Comment:
We are actually using message.logTime as reference for the event_window. Although this could change in the future, for the time being we should remain consistent.